craft_parts/plugins/go_plugin.py

```diff
@@ -120,12 +125,21 @@ def get_build_commands(self) -> list[str]:
         """Return a list of commands to run during the build step."""
         options = cast(GoPluginProperties, self._options)
 
+        init_cmds: list[str] = []
+        if options.go_workspace_use:
+            # Only initialize if go.work is not there.
+            init_cmds.append("[ -f go.work ] || go work init")
```

Comment:
I think it's non-trivial to support pre-existing go.work files, because:

- go will error out if you add the same module twice ("appears multiple times in workspace"). `go workspace use` will not check and silently do the right thing (i.e. not add the duplicate entry), afaict.
- their local go.work paths would likely not exist anyway (unless checked into the sourcecraft source tree -- which is not what we want them to do anyway)

So, we might need to explicitly error out here, at least for a start, because, without more intelligent parsing, we won't know if we're double adding a module.

Reply:
I was in conflict here as I do not want to potentially break existing projects (we are a rolling release after all :-D), but I guess they only break if they opt-in to the feature, for which it won't be a _no change break_, I'll make that amendment.